Uno.Gallery/Uno.Gallery.Shared/App.xaml.cs

```diff
@@ -226,28 +223,64 @@ public void SearchShellNavigateTo(Sample sample)
 			_shell.NavigationView.Content = page;
 		}
 
-		private Shell BuildShell()
+		private Shell BuildShell(LaunchActivatedEventArgs e)
 		{
 			_shell = new Shell();
 			AutomationProperties.SetAutomationId(_shell, "AppShell");
 			_shell.RegisterPropertyChangedCallback(Shell.CurrentSampleBackdoorProperty, OnCurrentSampleBackdoorChanged);
 			var nv = _shell.NavigationView;
 			AddNavigationItems(nv);
 
-			// landing navigation
-			ShellNavigateTo<OverviewPage>(
+			if (!IsThereSampleFilteredByArgs(nv, e))
+			{
+				// landing navigation
+				ShellNavigateTo<OverviewPage>(
 #if !WINDOWS
-				// workaround for uno#5069: setting NavView.SelectedItem at launch bricks it
-				trySynchronizeCurrentItem: false
+					// workaround for uno#5069: setting NavView.SelectedItem at launch bricks it
+					trySynchronizeCurrentItem: false
 #endif
-			);
+				);
+			}
 
 			// navigation + setting handler
 			nv.ItemInvoked += OnNavigationItemInvoked;
 
 			return _shell;
 		}
 
+		private bool IsThereSampleFilteredByArgs(MUXC.NavigationView nv, LaunchActivatedEventArgs e)
+		{
+#if __WASM__
+			var argumentsHash = Uno.Gallery.Wasm.FragmentHavigationHandler.CurrentFragment;
+#else
+			var argumentsHash = e.Arguments;
+#endif
```

Comment:
Is this idea ever used outside of html fragments in WASM? If not, wouldn't it be cleaner to wrap the entire method (and the call) in `#if __WASM__` instead of making it work for all platforms unnecessarily? There seems to be a lot of changes just to enable this outside of WASM, so just making sure we actually need it.

Reply:
@ramezgerges , 

I can't answer whether Gallery is used outside the address https://gallery.platform.uno.
Soon I implemented it for GTK to facilitate debug and have it ready in case it is necessary.
I believe that only Wasm serves the address gallery.platform.uno, but I couldn't say for sure.

Do you recommend removing?

Reply:
I'm not sure. It seems that the original issue was specific to WASM, but let's see what @agneszitte has to say.

Reply:
@ramezgerges 
I removed the changes to GTK in the commit below and left it only for Wasm. If you can review again.

https://github.com/unoplatform/Uno.Gallery/commit/6eecda4e98d3b761998be871698bdde40190ea81